File: 백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.java
import java.io.*;
import java.util.*;

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        
        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());
        
        String[] arr = new String[n];
        
        for(int i = 0; i < n; i++)
            arr[i] = br.readLine();
        
        int ans = 1000; //대충 많은 수로 설정
        for(int i = 0; i <= n - 8; i++) {
            for(int j = 0; j <= m - 8; j++) {
                int c = Check(arr, i , j);
                ans = Math.min(ans, c);
            }
        }
        System.out.println(ans);
    }
    public static int Check(String[] arr, int row, int col) {
        //체스판은 흑색으로 시작하거나 백색으로 시작함
        //우선, 흑색으로 시작하는 걸 기준으로 최소 개수를 구해보기
        String[] Chess = {"BWBWBWBW", "WBWBWBWB"};
        
        int cnt = 0;
        for(int i = 0; i < 8; i++) {
            for(int j = 0; j < 8; j++){
                if(arr[row + i].charAt(col + j) != Chess[i % 2].charAt(j))
                    cnt++;
            }
        }
        return Math.min(cnt, 64 - cnt);
    }
}

File: 백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.py
import sys;

n, m = map(int, sys.stdin.readline().strip().split())

board = []
ans = []

for i in range(n):
    board.append(sys.stdin.readline().strip())

for i in range(n - 7):
    for j in range(m - 7):
        wi = 0 # 시작점이 흰색일 경우 고쳐야하는 색 개수
        bi = 0 # 시작점이 검은색일 경우 고쳐야하는 색 개수
        for k in range(i, i + 8):
            for l in range(j, j + 8):
                if(k + l) % 2 == 0:  # 시작점과 같은 색이여야 하는 칸
                    if board[k][l] == 'B': # 시작점이 검은색일 경우
                        wi += 1 # 짝수번째 블럭이 흰색이라면 wi++
                    else:
                        bi += 1 # 시작점이 흰색인데 짝수번째이 흰색인 경우 bi++
                else: # 시작점과 반대인 색이여야 하는 칸
                    if board[k][l] == 'B': # 검은색일 경우
                        bi += 1 # 검은색 시작의 경우 짝수가 검은색이여야하므로
                    else: # 흰색일 경우
                        wi += 1 # 흰색 시작의 경우 짝수가 흰색이여야하므로
        ans.append(wi)
        ans.append(bi)

print(min(ans))
